Cartesian_kernel/include/CGAL/Cartesian/Reflection_rep_2.h

```diff
@@ -0,0 +1,211 @@
+    FT scal = l_to_v * Ox; //= |L|*cos(a)
+    FT det = l_to_v.y();// = |L|*sin(a)
+    sinus_ = 2*det*scal/l_to_v.squared_length(); //sin(2a) = 2*sin(a)*cos(a)
+    FT sq_cos = scal*scal/l_to_v.squared_length(); //cos(a)*cos(a)
+    cosinus_ = 2*sq_cos-1;
+  }
+
+  ~Reflection_repC2()
+  {}
+
+  Point_2      transform(const Point_2 &p) const
+  {
+    return Point_2(
+          cosinus_*p.x()+sinus_*p.y()-cosinus_*t.x()-sinus_*t.y()+t.x(),
+          sinus_*p.x()-cosinus_*p.y()-sinus_*t.x()+cosinus_*t.y()+t.y());
+  }
+
+  Vector_2      transform(const Vector_2 &p) const
+  {
+    return Vector_2(
+          cosinus_*p.x()+sinus_*p.y()-cosinus_*t.x()-sinus_*t.y()+t.x(),
+          sinus_*p.x()-cosinus_*p.y()-sinus_*t.x()+cosinus_*t.y()+t.y());
+  }
+
+  Direction_2  transform(const Direction_2 &d) const
+  {
+    
+    return transform(d.vector()).direction();
+  }
+
+  Aff_transformation_2 operator*(const Aff_t_base &t) const
+  {
+    return t.compose(*this);
+  }
+  
+  Aff_transformation_2 compose(const Translation &tr) const
+  {
+    return Aff_transformation_2(cosinus_, sinus_, t13()+tr.translationvector_.x(),
+                                sinus_, -cosinus_, t23()+tr.translationvector_.y());
+  }
+
+  Aff_transformation_2 compose(const Scaling &s) const
+  {
+    return Aff_transformation_2(s.scalefactor_ * cosinus_,
+                                s.scalefactor_ * sinus_,
+                                s.scalefactor_ * t13(),
+                                s.scalefactor_ * sinus_,
+                                -s.scalefactor_ * cosinus_,
+                                s.scalefactor_ * t23());
+  }
+
+  Aff_transformation_2 compose(const Transformation &tr) const
+  {
+    return Aff_transformation_2(
+          tr.t11*cosinus_+tr.t12*sinus_,
+          tr.t11*sinus_-tr.t12*cosinus_,
+          tr.t11*t13()+tr.t12*t23()+tr.t13,
+          tr.t21*cosinus_+tr.t22*sinus_,
+          tr.t21*sinus_-tr.t22*cosinus_,
+          tr.t21*t13()+tr.t22*t23()+tr.t23);
+  }
+  
+  Aff_transformation_2 compose(const Rotation &r) const
+  {
+    return Aff_transformation_2(
+          r.cosinus_*cosinus_-r.sinus_*sinus_,
+          r.cosinus_*sinus_+r.sinus_*cosinus_,
+          r.cosinus_*t13()
+          -r.sinus_*t23(),
+          r.sinus_*cosinus_+r.cosinus_*sinus_,
+          r.sinus_*sinus_-r.cosinus_*cosinus_,
+          r.sinus_*t13()
+          +r.cosinus_*t23());
+  }
+  
+  Aff_transformation_2 compose(const Reflection &r) const
+  {
+    return Aff_transformation_2(
+          cosinus_*r.cosinus_+sinus_*r.sinus_, 
+          r.cosinus_*sinus_-r.sinus_*cosinus_, 
+          r.cosinus_*(t13()-r.t.x()) + r.sinus_*(t23()-r.t.y())+r.t.x(),
+          
+          r.sinus_*cosinus_ - r.cosinus_*sinus_, 
+          r.sinus_*sinus_+r.cosinus_*cosinus_, 
+          r.sinus_*(t13()-r.t.x()) -r.cosinus_*(t23()-r.t.y())+r.t.y());
+  }
+  
+  Aff_transformation_2  inverse() const
+  {
+    return Aff_transformation_2(REFLECTION, line);
```

Comment:
The member `line` seems to be used only here, but the inverse of a reflection is itself. That means you can return `Aff_transformation_2(*this)`. Then there is no need, I think, to store the line in the transformation.

Reply:
It is not possible, `Reflection_rpC2` is not an `Aff_transformation_2`, so the constructor you are talking about doesn't exist.